File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    num_hash = {'zero':0, 'one':1, 'two':2, 'three':3, 'four':4, 
                'five':5, 'six':6, 'seven':7, 'eight':8, 'nine':9}
    for char, num in num_hash.items():
        if char in s:
            s = s.replace(char, str(num))
    return int(s)
    

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.swift
import Foundation

func solution(_ s:String) -> Int {
    var s = s
    let numDict = [0: "zero", 1: "one", 2: "two", 3: "three", 4: "four",
                   5: "five", 6: "six", 7: "seven", 8: "eight", 9: "nine"]
    
    for i in 0..<numDict.count {
        s = s.replacingOccurrences(of: numDict[i, default: ""], with: String(i))
    }
    
    guard let result = Int(s) else { return 0 }

    return result
}